.changes/next-release/bugfix-AWSDynamoDBEnhancedClient-f9d7822.json

```diff
@@ -0,0 +1,6 @@
+{
+    "type": "bugfix",
+    "category": "AWS DynamoDB Enhanced Client",
+    "contributor": "",
+    "description": "Publisher streams returned by async resources in the DynamoDB Enhanced Client now correctly handle mapping errors when they are encountered in the stream by calling onError on the subscriber and then implicitly cancelling the subscription. Previously the stream would just permanently hang and never complete."
```

Comment:
Just occurred to me, we should add a note about the fix in `SdkPublisher#map()` behavior under `AWS SDK for Java V2`